### ISSUE TITLE
Can't build on Debian 7

### ISSUE THREAD

**Issue by cimm** [hlandau/acme#34]
I am not a go programmer so it's probably a super n00b thing...

I installed the `golang` package via `apt` on Debian 7 and ran `make`. It ran for a while but stopped with the following error:

```
~/sources/acme$ make && sudo make install
	[RELOCATE]	  
	[GO-GET]	  github.com/hlandau/acme

src/gopkg.in/yaml.v2/decode.go:4:2: no Go source files in /usr/lib/go/src/pkg/encoding
make: *** [.gotten] Error 1
```

Any idea?

**Comment by hlandau** [comment]
Looks like Debian 7 is still at Go 1.0 (run `go version`) to check. The minimum requirement is Go 1.5.2 (or 1.5.0 if you don't intend to use the redirector; 1.5.0 and 1.5.1 have a bug affecting it.)

You might be better off using one of the [binary builds](https://github.com/hlandau/acme/releases). You just copy the binary to /usr/local/bin/acmetool (or wherever) and go from there.

**Comment by cimm** [comment]
Thank you for your quick reply. I installed the binary which is complaining as well:

```
./bin/acmetool: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.14' not found (required by ./bin/acmetool)
```

Guess the Debian 7 packages are too old to run newer software. Time to upgrade.

**Comment by kkofler** [comment]
The i386 version requires only GLIBC_2.3.4, the amd64 version wants memcpy@GLIBC_2.14.

See http://stackoverflow.com/questions/8823267/linking-against-older-symbol-version-in-a-so-file for a discussion of this issue.

It is generally a good idea to always build binaries on the OLDEST distribution possible (typically in a chroot). Building on a development machine (which tends to be fairly new) or on a too recent chroot is a bad idea, because of this one-way compatibility in glibc.

**Comment by kkofler** [comment]
Here is a hack that even @cimm as a user could do:
http://blog.marcus-brinkmann.de/2014/05/01/manual-symbol-version-override/

**Comment by hlandau** [comment]
Since the binaries are built by Travis I don't think there's much that can be done here. I could add a non-cgo build for Linux, though; all the non-Linux builds are non-cgo anyway. This only affects the redirector, and [only in these ways.](https://github.com/hlandau/service#reduced-functionality-mode)

**Comment by kkofler** [comment]
Looks like Travis uses Ubuntu Precise, which is the oldest version that Ubuntu still supports. It would be possible to work with a custom Docker container (https://docs.travis-ci.com/user/docker/), but then you'd have to take a container of something like CentOS 6 or even 5. (CentOS 5 is probably too old, it has the ancient libcap.so.1 that nothing else uses anymore, and it has no Go toolchain, but CentOS 6 has Go 1.5.1 available in EPEL, and it uses glibc 2.12.)

Are you sure that non-cgo builds won't be affected by this? memcpy is a very common symbol, I'm pretty sure that this comes from the Go runtime or compiler itself.

**Comment by hlandau** [comment]
Non-cgo builds are statically linked. I'll include them in the next release.

```$ ldd acmetool_linux_amd64_nocgo 
	not a dynamic executable```

**Comment by hlandau** [comment]
Non-cgo builds are now available, v0.0.20.
